### ISSUE TITLE
Metadata issue while building

### ISSUE THREAD

**Issue by RomaricS** [Teradata/covalent#985]
I'm reporting a Bug.
It seems like my covalent modules version is bumped from 3 to 4 when I try to run a build with the Jenkins job. I'm running ng build --prod and I'm getting this :
![image](https://user-images.githubusercontent.com/18727850/33066856-b7edbcd4-ceac-11e7-8aaf-80a1b9b981ac.png)

I'm not having the issue when doing it locally. I tried to delete the node_modules directory and to run npm install but it keeps bringing the version 4 in the metadata files. Here is my package.json file : 

`{
  "name": "dme-client",
  "version": "0.0.0",
  "license": "MIT",
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "build": "ng build",
    "test": "ng test",
    "lint": "ng lint",
    "e2e": "ng e2e",
    "postinstall": "ng build --prod"
  },
  "engines": {
    "node": ">= 7.4.0",
    "npm": ">= 4.0.5"
  },
  "private": true,
  "dependencies": {
    "@angular/animations": "^4.0.0",
    "@angular/cdk": "^2.0.0-beta.8",
    "@angular/cli": "1.1.2",
    "@angular/common": "^4.0.0",
    "@angular/compiler": "^4.0.0",
    "@angular/compiler-cli": "^4.0.0",
    "@angular/core": "^4.0.0",
    "@angular/flex-layout": "^2.0.0-beta.8",
    "@angular/forms": "^4.0.0",
    "@angular/http": "^4.0.0",
    "@angular/language-service": "^4.0.0",
    "@angular/material": "^2.0.0-beta.8",
    "@angular/platform-browser": "^4.0.0",
    "@angular/platform-browser-dynamic": "^4.0.0",
    "@angular/router": "^4.0.0",
    "@covalent/core": "^1.0.0-beta.5-1",
    "@covalent/dynamic-forms": "^1.0.0-beta.5-1",
    "@ngrx/core": "^1.2.0",
    "@ngrx/effects": "^2.0.3",
    "@ngrx/store": "^2.2.2",
    "@ngrx/store-devtools": "^3.2.4",
    "@ngx-translate/core": "^7.0.0",
    "@ngx-translate/http-loader": "^0.1.0",
    "@swimlane/ngx-charts": "^6.0.0",
    "angular-calendar": "^0.18.3",
    "angular2-csv": "^0.2.5",
    "angular2-moment": "^1.7.0",
    "core-js": "^2.4.1",
    "d3": "^4.9.1",
    "enhanced-resolve": "^3.3.0",
    "font-awesome": "^4.7.0",
    "hammerjs": "^2.0.8",
    "jspdf": "^1.3.3",
    "ng2-breadcrumb": "^0.5.14",
    "primeng": "^4.1.2",
    "print.js": "^1.0.18",
    "reselect": "^3.0.1",
    "rxjs": "^5.1.0",
    "zone.js": "^0.8.4"
  },
  "devDependencies": {
    "@types/jasmine": "2.5.45",
    "@types/node": "~6.0.60",
    "codelyzer": "~3.0.1",
    "jasmine-core": "~2.6.2",
    "jasmine-spec-reporter": "~4.1.0",
    "karma": "~1.7.0",
    "karma-chrome-launcher": "~2.1.1",
    "karma-cli": "~1.0.1",
    "karma-jasmine": "~1.1.0",
    "karma-jasmine-html-reporter": "^0.2.2",
    "karma-coverage-istanbul-reporter": "^1.2.1",
    "protractor": "~5.1.2",
    "ts-node": "~3.0.4",
    "tslint": "~5.3.2",
    "typescript": "~2.3.3"
  }
}`

**Comment by Gnorro** [comment]
same problem with ng build --prod

`{
  "name": "yeppon",
  "version": "0.0.0",
  "license": "MIT",
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "build": "ng build",
    "test": "ng test",
    "lint": "ng lint",
    "e2e": "ng e2e"
  },
  "private": true,
  "dependencies": {
    "@agm/core": "^1.0.0-beta.1",
    "@agm/js-marker-clusterer": "^1.0.0-beta.1",
    "@angular/animations": "^4.0.0",
    "@angular/cdk": "^2.0.0-beta.8",
    "@angular/common": "^4.0.0",
    "@angular/compiler": "^4.0.0",
    "@angular/core": "^4.0.0",
    "@angular/forms": "^4.0.0",
    "@angular/http": "^4.0.0",
    "@angular/material": "^2.0.0-beta.8",
    "@angular/platform-browser": "^4.0.0",
    "@angular/platform-browser-dynamic": "^4.0.0",
    "@angular/platform-server": "^4.3.1",
    "@angular/router": "^4.0.0",
    "@covalent/core": "^1.0.0-beta.5-1",
    "@types/angular": "^1.6.28",
    "braintree-web": "^3.22.2",
    "core-js": "^2.4.1",
    "enhanced-resolve": "^3.3.0",
    "hammerjs": "^2.0.8",
    "lory.js": "^2.3.3",
    "js-marker-clusterer": "^1.0.0",
    "moment": "^2.18.1",
    "ng2-cookies": "^1.0.12",
    "ng2-responsive": "^0.8.4",
    "ng2-scroll-to-el": "^1.0.0",
    "rxjs": "^5.1.0",
    "sass-loader": "^6.0.6",
    "web-animations-js": "^2.2.5",
    "zone.js": "^0.8.4"
  },
  "devDependencies": {
    "@angular/cli": "^1.4.7",
    "@angular/compiler-cli": "^4.0.0",
    "@angular/language-service": "^4.0.0",
    "@types/jasmine": "2.5.45",
    "@types/node": "~6.0.60",
    "codelyzer": "~3.0.1",
    "jasmine-core": "~2.6.2",
    "jasmine-spec-reporter": "~4.1.0",
    "karma": "~1.7.0",
    "karma-chrome-launcher": "~2.1.1",
    "karma-cli": "~1.0.1",
    "karma-coverage-istanbul-reporter": "^1.2.1",
    "karma-jasmine": "~1.1.0",
    "karma-jasmine-html-reporter": "^0.2.2",
    "ngx-swiper-wrapper": "^4.6.0",
    "protractor": "~5.1.2",
    "ts-node": "~3.0.4",
    "tslint": "~5.3.2",
    "typescript": "^2.5.3"
  }
}
`

**Comment by emoralesb05** [comment]
Please update to latest angular/material/covalent.. we dont support that beta release anymore since we are moving forward with the `RC` releases.. also since you both are using `^` in the dependencies, its hard to tell which dependencies you are really using.

**Comment by eciuca** [comment]
@emoralesb05 Did it work for you with the latest version? For me it doesn't ... I still get this error

**Comment by RomaricS** [comment]
I upgraded the modules,
Covalent to 1.0.0-rc.0
Material & Cdk to 5.0.0-rc0
All other Angular modules to 5.0.3
Typescript to 2.4.2
I had to change so many things due to material major changes but fit finally works.
If you are getting the mismatch error on angular packages make sure all of em are on the same version.
I did get it while upgrading modules while my @angular/core was still in V 4.0.3 and others like @angular/common were already at 5.0.3.
Hope it helps you..Cheers
